src/v/kafka/read_distribution_probe.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "config/configuration.h"
+#include "metrics/metrics.h"
+#include "utils/log_hist.h"
+
+#include <seastar/core/metrics.hh>
+
+#include <chrono>
+
+namespace kafka {
+class read_distribution_probe {
+public:
+    using hist_t = log_hist_read_dist;
+
+    read_distribution_probe() = default;
+    read_distribution_probe(const read_distribution_probe&) = delete;
+    read_distribution_probe& operator=(const read_distribution_probe&) = delete;
+    read_distribution_probe(read_distribution_probe&&) = delete;
+    read_distribution_probe& operator=(read_distribution_probe&&) = delete;
+    ~read_distribution_probe() = default;
+
+    template<typename dur_t>
+    void add_read_event_delta_from_tip(dur_t delta_from_tip) {
+        _read_distribution.record(delta_from_tip);
+    };
+
+    void setup_metrics() {
+        namespace sm = ss::metrics;
+
+        if (config::shard_local_cfg().disable_metrics()) {
+            return;
+        }
+
+        _metrics.add_group(
+          prometheus_sanitize::metrics_name("kafka:fetch"),
```

Comment:
While `:` is a valid metric name character `prometheus_sanitize::metrics_name` immediately changes it `_` so I don't think there is much point in having it being a colon in the first place.

(Know we do it in other places as well but always found it odd).

Reply:
Changed `:` to `_`.